lib/src/templates/template.dart

```diff
@@ -95,6 +95,39 @@ class FlutterPkgTemplate extends Template {
   }
 }
 
+/// {@template flutter_plugin_template}
+/// A Flutter plugin template.
+/// {@endtemplate}
+class FlutterPluginTemplate extends Template {
+  /// {@macro flutter_pkg_template}
+  FlutterPluginTemplate()
+      : super(
+          name: 'flutter_plugin',
+          bundle: flutterPluginBundle,
+          help: 'Generate a reusable Flutter plugin.',
+        );
+
+  @override
+  Future<void> onGenerateComplete(Logger logger, Directory outputDir) async {
+    final isFlutterInstalled = await Flutter.installed();
+    if (isFlutterInstalled) {
+      final installDependenciesDone = logger.progress(
+        'Running "flutter packages get" in ${outputDir.path}',
+      );
+      await Flutter.packagesGet(cwd: outputDir.path);
```

Comment:
```suggestion
      await Flutter.packagesGet(cwd: outputDir.path, recursive: true);
```

this needs to be recursive otherwise it'll fail the installation step.